### PR TITLE
Adding swagger spec for healthcareapis resource manager and examples.

### PR DESCRIPTION
**Pull request by YazanMSFT**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json [hunk 1]
```diff
@@ -0,0 +1,882 @@
...
+      }
+    },
+    "ServicesSkuInfo": {
+      "description": "Information about the SKU of the service instance.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the SKU.",
+          "enum": [
+            "Standard"
```
Comment (RyanBensonMSFT) [inline specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json]: Your resource doesn't have to have a sku, do you plan on adding more values in the future?

File: specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json [hunk 2]
```diff
@@ -0,0 +1,882 @@
...
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^[a-zA-Z0-9][a-zA-Z0-9-]{1,21}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "kind": {
```
Comment (RyanBensonMSFT) [inline specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json]: Are you using kind or etag? Any optional properties you aren't using should be removed.

File: specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json [hunk 3]
```diff
@@ -0,0 +1,882 @@
...
+          },
+          {
+            "$ref": "#/parameters/resourceName"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted - Delete request accepted; the operation will complete asynchronously."
+          },
+          "204": {
```
[the remark this answers is not in the document]
Comment (YazanMSFT) [reply specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json]: Updated the description.

**Review by anuchandy — CHANGES_REQUESTED** [review]
@YazanMSFT please add readme.md to the PR. here is an example: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/hanaonazure/resource-manager

**Comment by anuchandy** [comment]
@YazanMSFT could you take a look at my last comment?

**Comment by YazanMSFT** [comment]
Will do

File: specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json [hunk 4]
```diff
@@ -0,0 +1,982 @@
...
+      }
+  },
+  "paths": {
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HealthcareApis/services/{resourceName}": {
+          "get": {
+              "tags": [
+                  "Resource"
+              ],
+              "description": "Get the metadata of a service instance.",
+              "operationId": "Service_Get",
```
Comment (anuchandy) [inline specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json]: As mentioned over email please ensure operation group name is plural, so this should be Services_Get.

Comment (YazanMSFT) [reply specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2018-08-20-preview/healthcare-apis.json]: Thanks!

**Comment by YazanMSFT** [comment]
@anuchandy net SDK is failing with an error about not being able to handle a new RP. What needs to be done to resolve this?

**Comment by anuchandy** [comment]
@YazanMSFT I'm following with .NET team on this.

**Comment by anuchandy** [comment]
\\cc @shahabhijeet

**Comment by anuchandy** [comment]
sync-ed with @shahabhijeet on this. The process to onboard .NET sdks for new RP is in progress. For now we can consider it unblocked and new RPs will be manually onboarded.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**